plugin/PDFView.qml

```diff
@@ -178,7 +179,13 @@ SilicaFlickable {
         VerticalScrollDecorator { color: Theme.highlightDimmerColor; }
     ]
 
-    function goToPage(pageNumber) {
-        base.contentY = pdfCanvas.pagePosition( pageNumber );
+    function goToPage(pageNumber, top, left) {
+        // Adjust horizontal position if required.
+        if (left !== undefined && left >= 0.) {
+            base.contentX = base.itemWidth * left
```

Comment:
Tested with first pdf with links I could find from the internet: http://www.rpi.edu/dept/arc/docs/latex-thesis/latextopdf.pdf

On that, clicking "section 6" link on first page, the new position has text touching edges of the screen, which doesn't look nice. How about having some offset left and above new position? Could perhaps also guard document bottom to stay on the screen?